src/service_aggregator.py

```diff
@@ -408,21 +408,37 @@ async def filter_promiscuous_results(response,guid):
     MAX_C = 10
     if len(response["message"]["results"]) < MAX_C:
         return
-    prom_qnodes = get_promiscuous_qnodes(response)
+    prom_qnodes = await get_promiscuous_qnodes(response)
     #This is a dictionary from bound knodes to the index of their result
-    prom_counter = defaultdict(list)
     #There should only be one such node
     for qnode in prom_qnodes:
-        #How many distinct results have the same bozo in this spot?
+        # It's possible that there are multiple knodes that could be filtered.  But when we filter out the first one
+        # then the indices of the rest will change.  So we need to do this one at a time.
+        await remove_promiscuous_knodes(MAX_C, qnode, response)
+
+
+async def remove_promiscuous_knodes(MAX_C, qnode, response):
+    still_going = True
+    while still_going:
```

Comment:
I'm not sure this needs to be a `while` loop, at least at this point. We should only need to make the `prom_counter` once, right? Then we would just need to keep going over the prom_counter to make sure none of the knodes have too many results.

Reply:
it's possible that prom_counter could change if there are multiple bindings on the input qnode.  Not very likely though and maybe it'd be easier to read if we ignore that?  Leaving it for now I think.